inc/functions_post.php

```diff
@@ -401,64 +401,81 @@ function build_postbit($post, $post_type=0)
 		}
 
 		// Display profile fields on posts - only if field is filled in
-		if(is_array($profile_fields))
```

Comment:
If I have my plugin showing profile fields in announcements now I will need to replicate all the code.

As your code now assumes only the post bit for post should display profile fields.